content/docs/2.4/scalers/kubernetes-workload.md

```diff
@@ -0,0 +1,43 @@
++++
+title = "Kubernetes Workload"
+layout = "scaler"
+availability = "v2.4+"
+maintainer = "Community"
+description = "Scale applications based on the amount of pods which matches the given selectors."
+go_file = "kubernetes_workload_scaler"
++++
+
+### Trigger Specification
+
+```yaml
+triggers:
+- type: kubernetes-workload
+  metadata:
+    podSelector: 'app=backend'
+    namespace: 'default'
+    value: '1'
+```
+
+**Parameter list:**
+
+- `podSelector` - [Label selector](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors) that will be used to get the pod count. It supports multiple selectors split by coma (`,`). It also supports [set-based requirements](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#set-based-requirement) and a mix of them.
+- `namespace` - Namespace where the `podSelector` must be applied. If it's empty, all namespaces are eligibles. This parameter is optional and default value is empty.
```

Comment:
Shouldn't we be "secure by default" and only scan the same namespace when not specified? Just thinking out loud.

If we do, then we need to think how they would match cross namespace, but why would you do that since you typically target a given workload that you own, no?

Reply:
I'm not super sure about it, maybe you are right and we should scan only the own namespace as default but idk
From the security pov, maybe it could be better to use a specific service account which should be provided in order to control the permissions with granularity, but it could be more complex than now

Reply:
if you agree with it, I believe that I can make the change today :)

Reply:
That's a good point Tom. I think that we should limit this only to the current namespace (the one that the ScaledObject is being deployed to).

Reply:
In all cases or as default?

Reply:
All cases. We can enable this in the future, if there's demand.

Reply:
okey. I will update it :)